3.BinarySearch/search.js

```diff
@@ -0,0 +1,21 @@
+function binarySearch(arr, target) {
+  var half = 0;
+  var left = 0;
+  var right = arr.length - 1;
+  while(left <= right && target != arr[half]) {
```

Comment:
imo you could move
```
if (target == arr[half]) return half
```
under loop and remove `&& target != arr[half]` and `if` at the end of function